tests/functional/workloads/cnv/test_cnv_cluster_shutdown_recovery.py

```diff
@@ -0,0 +1,195 @@
+                else None
+            ),
+        )
+        all_vms.append(clone_obj)
+        csum = cal_md5sum_vm(vm_obj=clone_obj, file_path=file_paths[0])
+        source_csums[clone_obj.name] = csum
+
+        # Create a snapshot
+        # Taking Snapshot of PVC
+        pvc_obj = vm_for_snap.get_vm_pvc_obj()
+        snap_obj = snapshot_factory(pvc_obj)
+
+        # Restore the snapshot
+        res_snap_obj = snapshot_restore_factory(
+            snapshot_obj=snap_obj,
+            storageclass=vm_for_snap.sc_name,
+            volume_mode=snap_obj.parent_volume_mode,
+            access_mode=vm_for_snap.pvc_access_mode,
+            status=constants.STATUS_BOUND,
+            timeout=300,
+        )
+
+        # Create new VM using the restored PVC
+        res_vm_obj = cnv_workload(
+            source_url=constants.CNV_FEDORA_SOURCE,
+            storageclass=vm_for_snap.sc_name,
+            existing_pvc_obj=res_snap_obj,
+            namespace=vm_obj.namespace,
+        )
+        all_vms.append(res_vm_obj)
+        csum = cal_md5sum_vm(vm_obj=res_vm_obj, file_path=file_paths[0])
+        source_csums[res_vm_obj.name] = csum
+
+        # Keep vms in different states (power on, paused, stoped)
+        vm_for_stop.stop()
+        vm_for_snap.pause()
+
+        # Initiate abrupt shutdown the cluster nodes as per OCP official documentation
+        worker_nodes = get_nodes(node_type="worker")
+        master_nodes = get_nodes(node_type="master")
+
+        logger.info("Abruptly/Gracefully Shutting down worker & master nodes")
+        nodes.stop_nodes(nodes=worker_nodes, force=force)
+        nodes.stop_nodes(nodes=master_nodes, force=force)
+
+        logger.info("waiting for 5 min before starting nodes")
+        time.sleep(300)
+
+        # Initate ordered start of cluster after 10 min by following OCP official documentation.
+        logger.info("Starting worker & master nodes")
+        nodes.start_nodes(nodes=master_nodes)
+        nodes.start_nodes(nodes=worker_nodes)
+        retry(
+            (
+                CommandFailed,
+                TimeoutError,
+                AssertionError,
+                ResourceWrongStatusException,
+            ),
+            tries=30,
+            delay=15,
+        )(wait_for_nodes_status(timeout=1800))
+        logger.info("All nodes are now in READY state")
+
+        logger.info("Waiting for 10 min for all pods to come in running state.")
+        time.sleep(600)
+
+        # check cluster health
+        try:
+            logger.info("Making sure ceph health is OK")
+            Sanity().health_check(tries=50, cluster_check=False)
+        except Exception as ex:
+            logger.error("Failed at cluster health check!!")
+            raise ex
+
+        # CNV health check
+        cnv_obj = CNVInstaller()
+        cnv_obj.post_install_verification()
+
+        # Verify that VMs status post start
+        vm_obj.start()
+        vm_for_stop.start()
```

Comment:
Please add a check to confirm that the vm that was stopped before shutdown is still in stop state.

Reply:
Added